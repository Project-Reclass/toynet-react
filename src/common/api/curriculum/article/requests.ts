/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import axios from 'axios';

export interface ArticleImage {
  file: string;
  credit: string;

  caption?: string;
}

export interface ArticleMeta {
  source: string;
  title: string;
  author: string;
  images: ArticleImage[];
}

export const getArticleText = async (articleId: number): Promise<string> => {
  const { data } = await axios.get(`/data/article/${articleId}/article.md`);
  return data;
};

export const getArticleMeta = async (articleId: number): Promise<ArticleMeta> => {
  const { data } = await axios.get(`/data/article/${articleId}/meta.json`);
  return data;
};