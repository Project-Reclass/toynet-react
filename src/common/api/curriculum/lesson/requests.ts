import axios from 'axios';

type LessonSlideURI = string;
export interface Meta {
  numSlides: number;
}

export const getLessonSlide = async (submoduleId: number, slideId: number): Promise<LessonSlideURI> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/${slideId}.png`);
  return data;
};

export const getLessonMeta = async (submoduleId: number): Promise<Meta | undefined> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/meta.json`);
  return data;
};
