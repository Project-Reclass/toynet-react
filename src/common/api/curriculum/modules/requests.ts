import { simulateNetwork } from 'src/common/utils';
import modules from 'src/__data__/curriculum/modules.json';

export const getAllModules = () => {
  return simulateNetwork(() => modules);
};

export const getModuleById = (id: number) => {
  return simulateNetwork(() => modules.find(module => module.id === id));
};

export const getModuleByName = (name: string) => {
  return simulateNetwork(() => modules.find(module => module.name === name));
};