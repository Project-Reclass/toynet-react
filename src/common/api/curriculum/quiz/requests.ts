import axios from 'axios';

export const getQuizMeta = async (quizId: number) => {
	const { data } = await axios.get(`/data/quiz/${quizId}/meta.json`);
	return data;
};