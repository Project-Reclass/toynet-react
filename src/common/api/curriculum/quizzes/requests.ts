import { simulateNetwork } from 'src/common/utils';
import quizzes from 'src/__data__/curriculum/quizzes.json';

export const getAllQuizzes = async () => {
  return simulateNetwork(() => quizzes);
};

export const getQuizById = (id: number) => {
  return simulateNetwork(() => (
    quizzes.find(quiz => quiz.id === id)
  ));
};
