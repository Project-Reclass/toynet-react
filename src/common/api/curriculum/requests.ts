import { simulateNetwork } from 'src/common/utils';
import curriculums from 'src/__data__/curriculum/curriculums.json';

export const getAllCurriculums = () => {
  return simulateNetwork(() => curriculums);
};

export const getCurriculumById = (id: number) => {
  return simulateNetwork(() => (
    curriculums.find(value => value.id === id)
  ));
};