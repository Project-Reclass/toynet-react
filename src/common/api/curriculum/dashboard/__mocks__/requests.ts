/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/

import { DashboardIntf } from 'src/common/types/curriculum';

const defaultMockData = {
  id: 1,
  name: 'test',
  introduction: 'intro',
  modules: [],
};

export let mockData: DashboardIntf = {...defaultMockData};

export function __resetMock() {
  mockData = {...defaultMockData};
}

export function __setMockData(data: DashboardIntf) {
  mockData.id = data.id;
  mockData.name = data.name;
  mockData.introduction = data.introduction;
  mockData.modules = data.modules;
}

export const getCurriculum = async (): Promise<DashboardIntf | undefined> => {
  return Promise.resolve(mockData);
};