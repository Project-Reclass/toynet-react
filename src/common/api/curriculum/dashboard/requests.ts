import axios from 'axios';
import { DashboardIntf } from 'src/common/types/curriculum';

export const getCurriculum = async (curriculumId: number): Promise<DashboardIntf | undefined> => {
  const { data } = await axios.get(`/data/curriculum/${curriculumId}.json`);
  return data;
};