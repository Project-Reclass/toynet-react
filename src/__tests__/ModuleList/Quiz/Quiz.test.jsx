import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { waitFor } from '@testing-library/react';

import Quiz from 'src/Curriculum/Quiz';

jest.mock('src/common/api/curriculum/quiz/requests');
import { getQuizMeta } from 'src/common/api/curriculum/quiz/requests';
import { renderWithTheme } from 'src/common/test-utils/renderWithTheme';

const RenderWithRouter = ({ children, moduleId, quizId }) => (
  <MemoryRouter initialEntries={[`/module/${moduleId}/quiz/${quizId}`]}>
    <Route path="/module/:moduleId/quiz/:quizId">{children}</Route>
  </MemoryRouter>
);

describe('The Quiz page', () => {
  it('should render the same based on URL parameters', () => {
    const { container } = renderWithTheme(
      <RenderWithRouter moduleId={42} quizId={64}>
        <Quiz />
      </RenderWithRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should fetch quiz data', async () => {
    const data = [
      {
        "question": "How many nodes share a single channel on a bus topology?",
        "options": [
            "One node",
            "Only ones communicating",
            "Three nodes",
            "All nodes"
        ],
        "answer": 3
      },
      {
        "question": "Which of the following is the most fault-tolerant WAN topology?",
        "options": [
            "Partial Mesh",
            "Ring",
            "Star",
            "Full-mesh"
        ],
        "answer": 3
      }
    ];
    getQuizMeta.mockResolvedValue({ items: data });
    const { getByText, getAllByText } = renderWithTheme(
      <RenderWithRouter quizId={64}>
        <Quiz />
      </RenderWithRouter>
    );

    await waitFor(() => getAllByText(/How many nodes/i));

    expect(getByText(/How many nodes/i)).toBeInTheDocument();
    expect(getByText(/fault-tolerant WAN topology/i)).toBeInTheDocument();
    expect(getByText(/Partial Mesh/i)).toBeInTheDocument();
  });
});