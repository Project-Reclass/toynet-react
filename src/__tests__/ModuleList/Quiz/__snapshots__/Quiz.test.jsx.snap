// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Quiz page should render the same based on URL parameters 1`] = `
<div
  className="css-1cxsl7j"
>
  <div
    className="css-3pksgp"
  />
  <div
    className="css-k008qs"
  >
    <a
      className="css-jiw0x4"
      onClick={[Function]}
    >
      Submit Quiz
    </a>
  </div>
</div>
`;
